### ISSUE TITLE
setting the query to the same key should overwrite the other on an agent

### ISSUE THREAD

**Issue by Jordan4jc** [visionmedia/superagent#1395]
I created an agent and then later I set the query

```
export const API = request.agent()
  .use(process.env.NODE_ENV === 'production' ? prefix('https://api.***') : prefix('https://staging.api.***'));
```

Sometime later...

`API.use(req => req.query({ company_id: persistedState.companies.activeCompany}))`

Then when the user interacts with the system in some way to change that `activeCompany` property, I want to change the query string to that new value, so I do the same thing but instead of it changing to the new value I see both values as query string parameters attached to all my API calls.

If this is the intended effect, is there a way to remove the query string that was set before I set it to the new one?

**Comment by kornelski** [comment]
So, we have this huge dilemma between perfectly preserving formatting of existing query string, and updating it (some people are very picky about their query strings and want them exactly as set, byte-identical, in the right order, with the specific encoding of specific characters).

To keep everyone happy, superagent keeps query string both as a whole string, and also as a separate object with keys/values, and then tries to merge them. And we used to have separate implementations for browsers and Node, because they use different libraries.

The behavior you're expecting seems reasonable. I suppose we did something wrong with merging the strings?
